src/DIRAC/WorkloadManagementSystem/Client/PilotLoggingPlugins/BasicPilotLoggingPlugin.py

```diff
@@ -0,0 +1,50 @@
+"""
+Basic Pilot logging plugin. Just log messages.
+"""
+from DIRAC import S_OK, S_ERROR, gLogger
+
+sLog = gLogger.getSubLogger(__name__)
+
+
+class BasicPilotLoggingPlugin:
```

Comment:
Maybe it's time to use abstract classes?

```suggestion
from abc import ABC

class PilotLoggingPlugin(ABC):
```
and so on?

Here I'd also rename from *BasicPilotLoggingPlugin* to *PilotLoggingPlugin*

Reply:
Well, I was a bit sceptical about this. An abstract class cannot be instantiated, and we are loading a base class if someone enables remote logging but forgets to specify a plugin:
[https://github.com/martynia/DIRAC/blob/6c280fed14ad6439c587bcdc14ab32fb9c951a4f/src/DIRAC/WorkloadManagementSystem/Service/TornadoPilotLoggingHandler.py#L34
](https://github.com/martynia/DIRAC/blob/6c280fed14ad6439c587bcdc14ab32fb9c951a4f/src/DIRAC/WorkloadManagementSystem/Service/TornadoPilotLoggingHandler.py#L34)

Then at least the server loads a default and logs from  `sendMessage`. If you prefer `Failed to load LoggingPlugin` message instead then we could create an abstract class.

Reply:
Done with abstract classes.